### DISCUSSION TITLE
Increase the number of concurrencies of tasmota tcp bridge and compile

### DISCUSSION THREAD

**Discussion by wei1980okb** [arendst/Tasmota#20092] [Ideas]
https://tasmota.github.io/docs/Serial-to-TCP-Bridge/
I am testing the tasmota tcp bridge function. Check the above document. There are only 2 concurrent connections by default. Now I need to add more concurrent connections. Through searching, I found that tasmota 9.5.0 has the following parameters.
TCPBRIDGE_MAX_CLIENTS
This is where the maximum number of connections for the TCP Bridge is specified. Change it to your desired value,
Add the following lines to user_config_override.h and compile, you can have 4 concurrent
#define TCPBRIDGE_MAX_CLIENTS 4
But in the current docs, I can't find how to modify the concurrency parameters. And please tell me the format and tutorial for adding this parameter compilation now!
I need your help, thank you!

**Comment by sfromis** [comment]
What I see in current Tasmota (and also the badly outdated 9.5.0), is instead `#define TCP_BRIDGE_CONNECTIONS`. Not sure where you found what you post about having 4 concurrent, and not clear what you mean by "concurrency parameter**s**".

As the docs page you link explains, `user_config_override.h` is where you can add build time configs, including `#define TCP_BRIDGE_CONNECTIONS 4` if you want to try that. I don't *know* if 4 is actually supported, or how much have been tested, but it does appear that the code in the driver was changed from having connection 1 and 2 to instead have a array of connections, which does imply more than 2.

**Comment by wei1980okb** [comment]
Thank you for your reply, I have compiled and tested it and it works fine:
Test 1 (cannot use code)
Add in user_config_override.h
#ifdef USE_TCP_BRIDGE
#undef USE_TCP_BRIDGE
#endif
#define TCP_BRIDGE_CONNECTIONS 10
Then after compilation, there are no TCP RX and TCP TX options in the tasmota webgui module settings. Note that this command cannot be used normally.

Test 2 (the code can be used normally)
Add in user_config_override.h
#ifdef USE_TCP_BRIDGE
#undef USE_TCP_BRIDGE
#endif
#define TCP_BRIDGE_CONNECTIONS 10
#define USE_TCP_BRIDGE
Then after compilation, there are TCP RX and TCP TX options in the module settings. The number of concurrent connections can reach 10. When there is an 11th connection, the first connection will be automatically disconnected.

result
After passing the second test, it can meet my requirements normally. Thank you very much.
